core/src/main/scala/com/fortysevendeg/mvessel/statement/PreparedStatement.scala

```diff
@@ -13,9 +14,9 @@ import com.fortysevendeg.mvessel.util.StreamUtils._
 
 import scala.util.{Failure, Success, Try}
 
-class PreparedStatement(
+class PreparedStatement[T <: CursorProxy](
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Number of methods in class exceeds 30](https://www.codacy.com/app/47deg/mvessel/file/970159414/issues/source?bid=2442814&fileBranchId=2859700#l17)